### DISCUSSION TITLE
Tyler Krupicka: Stripe TypeScript Migration

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#82] [Announcements]
## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

Justin: Hey everyone! Today we're joined by Tyler Krupicka. Tyler's on the developer experience team at Stripe and we're excited to have him join us to talk about his role and his recent experience helping migrate Stripe's codebase from Flow to TypeScript. Before we get into that though, Tyler would you like to tell our audience a little more about yourself?

### Developer productivity

- [x] You work on the developer experience team at Stripe. What does a team like that typically do at a large company?
- [x] What are some examples (other than the typescript migration) of initiatives this team has driven?

### TypeScript Migration

- [x] Why did the company choose to migrate from flow to typescript? 
- [x] How did you all go about migrating so many lines of code? What options were you choosing between in how to approach it?
- [x] Is the codemod perfect? Or does it bail out and use `any` sometimes? Does it add any `TODO` types?
- [x] Were there any road bumps with working on a Typescript repo of such magnitude?
- [x] How has the rollout gone?
- [x] Do you all plan to open source the tool?

### Future

- [x] What are the next frontiers of developer experience that are going to be conquered by the team you work on?
- [x] What new developer tools are you most excited about?

## Outro

Andrew: Well that’s it for this week’s episode of devtools.fm. Be sure to follow us on YouTube and wherever you consume your podcasts. Thanks for listening!

**Comment by tylerkrupicka** [comment]
Here's a [Twitter thread](https://twitter.com/alunny/status/1501261144341680130) outlining some of the high level points about the migration.

Might be interesting to talk about Flow, our initial planning, how we got started on the codemod, planning the 3.5 million line codebase, the migration, and progress since migrating.

**Comment by hipstersmoothie** [comment]
My tooltips:

- https://virtuoso.dev
- https://github.com/ecklf/tailwindcss-radix

**Comment by tylerkrupicka-stripe** [comment]
Tooltips:

- https://www.npmjs.com/package/eslint-bankruptcy
- https://github.com/dsherret/ts-morph

**Reply by just-be-dev** [reply]
Tooltips:
- https://riffle.systems/essays/prelude/
- https://github.com/maraisr/meros

**Comment by tylerkrupicka-stripe** [comment]
For the show notes:

- [Blog post](https://stripe.com/blog/migrating-to-typescript)
- [OSS conversion tool](https://github.com/stripe-archive/flow-to-typescript-codemod)
- [Stripe engineering blog](https://stripe.com/blog/engineering)
